### ISSUE TITLE
Running installer after installing win git

### ISSUE THREAD

**Issue by pschloss** [swcarpentry/windows-installer#42]
Getting ready for an upcoming workshop, I thought it would be good to try and run some of the material in windows. Having a mac handy, I'm trying to run the installer using a parallel's version of windows, so feel free to tell me that's my problem... I already had win git installed from git-scm.com. Then I ran the SWC installer and it went through various security options and made it look like it was about to do it's thing when a black terminal window opened and closed fairly quickly. I suspected that having git installed already would cause problems, so I deleted the various git folders and tried again with the SWC installer. Same result. Do you know what might be going wrong? I don't see any icons like the instructions suggest should be on my desktop.

**Comment by wking** [comment]
On Wed, Feb 24, 2016 at 07:25:31AM -0800, Pat Schloss wrote:
> … it was about to do it's thing when a black terminal window opened
> and closed fairly quickly…

I think that's what it looks like when it works, see #13.  I'm not
sure what @ethanwhite's “proper [cues]” were [1], but maybe we need to
make them more prominent?

[1]: https://github.com/swcarpentry/windows-installer/pull/13#issuecomment-48673766

**Comment by ethanwhite** [comment]
Having git installed shouldn't be a problem and I don't see why running in parallels should be an issue as long as Windows can get to the internet.

Following up on @wking's point - let's figure out whether it actually worked or not. If you open Git Bash can run `nano` does it work? Do you have a `.swc` directory in your home folder?

**Comment by pschloss** [comment]
Great - thanks for the help! `nano` does not work and there is a `.swc` directory. Looking through the thread at #32 there's no `.bash_profile` in my home directory, which is `/c/Users/pschloss/`

**Comment by ethanwhite** [comment]
OK, that's helpful (albeit confusing).

Is `nano` not found or is there a different error?

If you have Python installed can you get rid of the current `.swc` directory and then run https://github.com/swcarpentry/windows-installer/blob/master/swc-windows-installer.py using the verbose option:

`python swc-windows-installer.py --verbose`

and post the output.

**Comment by ethanwhite** [comment]
> a black terminal window opened and closed fairly quickly

this would seem to indicate that there could be an issue early in the process because this would normally take 10-15 seconds and display text about downloading/installing different things. Running the .py file directly should help show us where the error is showing up.

**Comment by pschloss** [comment]
Sorry for the delay - I had to install python...

```
pschloss@PATSCHLOSS7560 MINGW64 ~
$ python swc-windows-installer.py --verbose error

pschloss@PATSCHLOSS7560 MINGW64 ~
$ python swc-windows-installer.py --verbose critical

pschloss@PATSCHLOSS7560 MINGW64 ~
$ python swc-windows-installer.py --verbose warning

pschloss@PATSCHLOSS7560 MINGW64 ~
$ python swc-windows-installer.py --verbose info
Preparing your Software Carpentry awesomeness!
installer version 0.3
create nosetests entrypoint C:\Users\pschloss\.swc\bin\nosetests
existing installation at C:\Users\pschloss\.swc\lib\make
existing installation at C:\Users\pschloss\.swc\lib\make
existing installation at C:\Users\pschloss\.swc\lib\nano
existing installation at C:\Users\pschloss\.swc\share\nanorc
existing installation at C:\Users\pschloss\.swc\lib\sqlite
no R installation found under C:\Program Files (x86)
update bash profile at C:\Users\pschloss\.bash_profile
Installation complete.

pschloss@PATSCHLOSS7560 MINGW64 ~
$ python swc-windows-installer.py --verbose debug
Preparing your Software Carpentry awesomeness!
installer version 0.3
create nosetests entrypoint C:\Users\pschloss\.swc\bin\nosetests
existing installation at C:\Users\pschloss\.swc\lib\make
existing installation at C:\Users\pschloss\.swc\lib\make
existing installation at C:\Users\pschloss\.swc\lib\nano
existing installation at C:\Users\pschloss\.swc\share\nanorc
existing installation at C:\Users\pschloss\.swc\lib\sqlite
no R installation found under C:\Program Files (x86)
update bash profile at C:\Users\pschloss\.bash_profile
extra paths:
* C:\Users\pschloss\.swc\lib\make\bin
* C:\Users\pschloss\.swc\lib\nano
* C:\Users\pschloss\.swc\lib\sqlite
* C:\Users\pschloss\.swc\bin
Installation complete.
```

Reading through this, it looks like the `--verbose info` flag created the `.bash_profile` file and once I closed and reopened the terminal, nano was working.

**Comment by ethanwhite** [comment]
The `--verbose` flag shouldn't create anything, it's just reporting different information back.

If you delete `.swc` and `.bash_profile` and then run:

`python swc-windows-installer.py`

I suspect that everything will work.

Please confirm and then we can try to figure out why the installer didn't actually do this (since that's basically all it does).

**Comment by pschloss** [comment]
That worked

**Comment by ethanwhite** [comment]
OK, so now can you remove `.swc` and `.bash_profile`, click on the start menu, and search for and the run `Software Carpentry Windows Installer`. Let me know if that gives you `.swc` and `.bash_profile` and if nano works. Thanks for helping debug this.

**Comment by pschloss** [comment]
That worked too.

**Comment by ethanwhite** [comment]
Weird. OK, let's confirm that the first thing still isn't working. So, remove `.swc` and `.bash_profile`, uninstall the installer by going to Start -> Control Panel -> Uninstall a program and uninstalling the Software Carpentry Windows Installer. Download a fresh copy of the installer and run it again. Thanks.

**Comment by pschloss** [comment]
That worked too ... I think I totally misinterpreted how this works ... I didn't have python installed since this is going to be a R-based workshop and I now realize that the installer uses python scripts. If that's the case then I'm sorry for taking you down the rabbit hole. So are we assuming that users will install Python even if they are in an R-based workshop?

**Comment by ethanwhite** [comment]
No, you didn't misinterpret anything. The installer packages it's own version of Python, which is what it uses. You should be able to uninstall Python and rerun everything and it should work exactly the same way. Go ahead and try that to make sure, but my guess is there was just some sort of weird hiccup on your initial install based on... I'm not sure, maybe a temporarily bad internet connection or something.

If things work after you've uninstalled Python & removed `.swc` and `.bash_profile` then feel free to go ahead and close this. If it stops working after uninstalling Python then I'll be really confused and we'll keep debugging.

**Comment by wking** [comment]
On Wed, Feb 24, 2016 at 10:51:46AM -0800, Ethan White wrote:
> … my guess is there was just some sort of weird hiccup on your
> initial install based on... I'm not sure, maybe a temporarily bad
> internet connection or something.

That sounds likely to me as well.

> If things work after you've uninstalled Python & removed `.swc` and
> `.bash_profile` then feel free to go ahead and close this.

Or we can leave it open until we figure out a way to have Inno Setup
at least check the exit status of the Python script and ideally
display it's stderr somewhere visible.  Having a weird hiccup disrupt
installation is unlikely, but covering that corner case with a UI
that's as friendly as possible will make this nicer for novice
leaners.

**Comment by ethanwhite** [comment]
> Or we can leave it open until we figure out a way to have Inno Setup at least check the exit status of the Python script and ideally display it's stderr somewhere visible. Having a weird hiccup disrupt installation is unlikely, but covering that corner case with a UI that's as friendly as possible will make this nicer for novice leaners.

That's certainly a good plan. I suspect it's going to be quite difficult (and to be honest I'm not sure if it will be possible given the way things are set up in general). If everything works for @pschloss I'm happy to leave this open in case someone has the time to try to make this work.

**Comment by pschloss** [comment]
Hey everyone, thanks for the help - I'm not sure what happened, but everything works after I remove the installer, python, .bash_profile and .swc and reinstall. I vote for the hiccup hypothesis, but will leave it to you whether to close the issue.

**Comment by ethanwhite** [comment]
Thanks @pschloss. Good luck with the workshop!
